examples/mysql-galera/image/docker-entrypoint.sh

```diff
@@ -0,0 +1,121 @@
+      exit 1
+    fi
+
+    echo 'Running mysql_install_db ...'
+        mysql_install_db --datadir="$DATADIR"
+        echo 'Finished mysql_install_db'
+
+    
+    # These statements _must_ be on individual lines, and _must_ end with
+    # semicolons (no line breaks or comments are permitted).
+    # TODO proper SQL escaping on ALL the things D:
+    
+    tempSqlFile='/tmp/mysql-first-time.sql'
+    cat > "$tempSqlFile" <<-EOSQL
+DELETE FROM mysql.user ;
+CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
+GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
+DROP DATABASE IF EXISTS test ;
+EOSQL
+    
+    if [ "$MYSQL_DATABASE" ]; then
+      echo "CREATE DATABASE IF NOT EXISTS \`$MYSQL_DATABASE\` ;" >> "$tempSqlFile"
+    fi
+    
+    if [ "$MYSQL_USER" -a "$MYSQL_PASSWORD" ]; then
+      echo "CREATE USER '$MYSQL_USER'@'%' IDENTIFIED BY '$MYSQL_PASSWORD' ;" >> "$tempSqlFile"
+      
+      if [ "$MYSQL_DATABASE" ]; then
+        echo "GRANT ALL ON \`$MYSQL_DATABASE\`.* TO '$MYSQL_USER'@'%' ;" >> "$tempSqlFile"
+      fi
+    fi
+
+    if [ -n "$GALERA_CLUSTER" -a "$GALERA_CLUSTER" = true ]; then
+      WSREP_SST_USER=${WSREP_SST_USER:-"sst"}
+      if [ -z "$WSREP_SST_PASSWORD" ]; then
+        echo >&2 'error: database is uninitialized and WSREP_SST_PASSWORD not set'
+        echo >&2 '  Did you forget to add -e WSREP_SST_PASSWORD=xxx ?'
+        exit 1
+      fi
+
+      sed -i -e "s|wsrep_sst_auth \= \"sstuser:changethis\"|wsrep_sst_auth = ${WSREP_SST_USER}:${WSREP_SST_PASSWORD}|" /etc/mysql/conf.d/cluster.cnf
+
+      WSREP_NODE_ADDRESS=`ip addr show | grep -E '^[ ]*inet' | grep -m1 global | awk '{ print $2 }' | sed -e 's/\/.*//'`
+      if [ -n "$WSREP_NODE_ADDRESS" ]; then
+        sed -i -e "s|^#wsrep_node_address \= .*$|wsrep_node_address = ${WSREP_NODE_ADDRESS}|" /etc/mysql/conf.d/cluster.cnf
+      fi
+      
+      if [ -z "$WSREP_CLUSTER_ADDRESS" -o "$WSREP_CLUSTER_ADDRESS" == "gcomm://" ]; then
+        if [ -z "$WSREP_CLUSTER_ADDRESS" ]; then
+          WSREP_CLUSTER_ADDRESS="gcomm://"
+        fi
+        for NUM in `seq 1 $NUM_NODES`;
+        do
+          NODE="PXC_NODE${NUM}_SERVICE_HOST"
+      
+          if [ -n "${!NODE}" ]; then
+            # if not itself, then add
+            if [ $(expr "$HOSTNAME" : "pxc-node${NUM}") -eq 0 ]; then
+              if [ $WSREP_CLUSTER_ADDRESS != "gcomm://" ]; then
+                WSREP_CLUSTER_ADDRESS="${WSREP_CLUSTER_ADDRESS},"
+              fi
+              WSREP_CLUSTER_ADDRESS="${WSREP_CLUSTER_ADDRESS}"${!NODE}
+            fi
+          fi
+        done
+      fi
+  
+      # Ok, now that we went through the trouble of building up a nice
+      # cluster address string, regex the conf file with that value 
+      if [ -n "$WSREP_CLUSTER_ADDRESS" -a "$WSREP_CLUSTER_ADDRESS" != "gcomm://" ]; then
+        sed -i -e "s|wsrep_cluster_address \= gcomm://|wsrep_cluster_address = ${WSREP_CLUSTER_ADDRESS}|" /etc/mysql/conf.d/cluster.cnf
```

Comment:
(nit) I think the spaces around = are optional: `wsrep_cluster_address ?\= ?gcomm://`